stores/sql/main.go

```diff
@@ -192,20 +192,16 @@ func ArchiveContract(ctx context.Context, tx sql.Tx, fcid types.FileContractID,
 		return fmt.Errorf("failed to delete contract_sectors: %w", err)
 	}
 
-	// delete host sectors that are no longer associated with any contract
+	// delete all host_sectors for every host that we don't have an active
+	// contract with anymore
 	_, err = tx.Exec(ctx, `DELETE FROM host_sectors
-WHERE NOT EXISTS (
-    SELECT 1
-    FROM contract_sectors
-    WHERE contract_sectors.db_sector_id = host_sectors.db_sector_id
-)
-OR NOT EXISTS (
-    SELECT 1
-    FROM contracts
-    INNER JOIN hosts ON contracts.host_id = hosts.id
-    WHERE contracts.archival_reason IS NULL
-    AND hosts.id = host_sectors.db_host_id
-)`)
+		WHERE NOT EXISTS (
```

Comment:
Had to change this so that the migration succeeds. Otherwise we'd just immediately delete all host sectors even though we still need them for migrations.